### PR TITLE
add spidermonkey 1.8.5

### PR DESCRIPTION
**Pull request by grueni**

### CODE REVIEW

File: components/runtime/spidermonkey/Makefile [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# Copyright 2017 Andreas Grueninger, Grueninger GmbH, (grueni). All rights reserved. 
+#
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=	          spidermonkey
+COMPONENT_VERSION=        1.8.5
+COMPONENT_SUMMARY=        Javascript engine
+COMPONENT_PROJECT_URL=    https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey
+COMPONENT_FMRI=	          runtime/spidermonkey
```
Comment (pyhalov) [inline components/runtime/spidermonkey/Makefile]: library/spidermonkey (as it's not an interpreter), please move the component correspondingly

File: components/runtime/spidermonkey/Makefile [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+COMPONENT_CLASSIFICATION= Development/Other Languages
```
Comment (pyhalov) [inline components/runtime/spidermonkey/Makefile]: System/Library

**Comment by pyhalov** [comment]
Do we need Studio and SPARC-specific patches?

**Comment by grueni** [comment]
We don't need it. But we are gently guys and maybe other people (sparc!) are interested.
Where did you find the Studio-specific patches?
I thought I commented them out.

File: components/runtime/spidermonkey/patches/02-jsfunc.patch [hunk 3]
```diff
@@ -0,0 +1,12 @@
+diff --git a/js/src/jsfun.h b/js/src/jsfun.h
+index a52092c..9555a03 100644
+--- a/jsfun.h
++++ b/jsfun.h
+@@ -144,6 +144,7 @@ struct JSFunction : public JSObject_Slots2
+         void            *nativeOrScript;
+     } u;
+     JSAtom          *atom;        /* name for diagnostics and decompiling */
++    void            *dummy;       /* for building with Solaris Studio */
```
Comment (pyhalov) [inline components/runtime/spidermonkey/patches/02-jsfunc.patch]: seems to be relevant for studio only (based on comment). Is it true?

File: components/runtime/spidermonkey/patches/05-pgo-ss12_2.patch [hunk 4]
```diff
@@ -0,0 +1,18 @@
+diff --git a/js/src/configure b/js/src/configure
+index 0e98daf..11bb34f 100755
+--- a/configure
++++ b/configure
+@@ -14560,10 +14560,10 @@ if test -n "$INTEL_CC"; then
+ fi
+ 
+ if test "$SOLARIS_SUNPRO_CC"; then
+-  PROFILE_GEN_CFLAGS="-xprofile=collect:$_objdir/$enable_application"
```
Comment (pyhalov) [inline components/runtime/spidermonkey/patches/05-pgo-ss12_2.patch]: Relevant only for Studio

File: components/library/spidermonkey/Makefile [hunk 5]
```diff
@@ -18,7 +18,7 @@ COMPONENT_NAME=	          spidermonkey
 COMPONENT_VERSION=        1.8.5
 COMPONENT_SUMMARY=        Javascript engine
 COMPONENT_PROJECT_URL=    https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey
-COMPONENT_FMRI=	          runtime/spidermonkey
+COMPONENT_FMRI=	          library/javascript/spidermonkey
```
Comment (pyhalov) [inline components/library/spidermonkey/Makefile]: as I understand, Oracle took library/javascript/* for JS libraries. Let's use just library/spidermonkey, as it's C library for JS processing :)

File: components/runtime/spidermonkey/patches/05-pgo-ss12_2.patch (hunk 4 above)
Comment (grueni) [reply components/runtime/spidermonkey/patches/05-pgo-ss12_2.patch]: It is protected so we can use the patch or remove it. It doesn't matter.

File: components/runtime/spidermonkey/patches/02-jsfunc.patch (hunk 3 above)
Comment (grueni) [reply components/runtime/spidermonkey/patches/02-jsfunc.patch]: I tested with the patch and all tests passed successfully.
So I would use it.

**Comment by grueni** [comment]
Ok, it is now in component/library.

**Comment by alarcher** [comment]
@pyhalov @grueni Unfortunately it does not build with gcc-6, probably because 1.8.5 is an outdated version. Are we able to use at least version 45 from https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey/Releases ?

**Comment by grueni** [comment]
Unfortunately no.  In the issues section of the CouchDB project the team explains why. They use anonymous functions which are not available with younger versions of couchdb.

**Pull request merged**